numba/core/bytecode.py

```diff
@@ -321,45 +361,199 @@ def get_used_globals(self):
 
 
 def _fix_LOAD_GLOBAL_arg(arg):
-    if PYVERSION >= (3, 11):
-        assert PYVERSION == (3, 11) # reminder to check newer versions
+    if PYVERSION in ((3, 11), (3, 12)):
         return arg >> 1
-    return arg
+    elif PYVERSION in ((3, 8), (3, 9), (3, 10)):
+        return arg
+    else:
+        raise NotImplementedError(PYVERSION)
 
 
 class ByteCodePy311(_ByteCode):
+
     def __init__(self, func_id):
         super().__init__(func_id)
-
-        def fixup_eh(ent):
-            from dis import _ExceptionTableEntry
-            # Patch up the exception table offset
-            # because we add a NOP in _patched_opargs
-            out = _ExceptionTableEntry(
-                start=ent.start + _FIXED_OFFSET, end=ent.end + _FIXED_OFFSET,
-                target=ent.target + _FIXED_OFFSET,
-                depth=ent.depth, lasti=ent.lasti,
-            )
-            return out
-
         entries = dis.Bytecode(func_id.code).exception_entries
-        self.exception_entries = tuple(map(fixup_eh, entries))
+        self.exception_entries = tuple(map(self.fixup_eh, entries))
+
+    @staticmethod
+    def fixup_eh(ent):
+        # Patch up the exception table offset
+        # because we add a NOP in _patched_opargs
+        out = dis._ExceptionTableEntry(
+            start=ent.start + _FIXED_OFFSET, end=ent.end + _FIXED_OFFSET,
+            target=ent.target + _FIXED_OFFSET,
+            depth=ent.depth, lasti=ent.lasti,
```

Comment:
Does `lasti` also need moving by `_FIXED_OFFSET`?

Reply:
Perhaps, I do wonder if this is even used by Numba. Perhaps there is a way to trigger a bug too?

Reply:
After an OOB discussion with @stuartarchibald and @kc611 - we came to the conclusion, that Numba probably doesn't use `lasti` -- but that doesn't mean we shouldn't be consistent. The consensus was to simply add the `_FIXED_OFFSET`. 

There is also some more information here:

https://discuss.python.org/t/when-do-you-set-lasti-in-exception-tables/22021/6

Reply:
added via 907a60471496c81374d88ac4c91eb4e256f546c5

Reply:
I had to unresolve this conversation, as it turns out this change causes a significant number of failures across 3.11 and 3.12:

https://dev.azure.com/numba/numba/_build/results?buildId=16289&view=logs&jobId=0e486a9e-8d4d-585c-1131-b2203bf73f0e&j=0e486a9e-8d4d-585c-1131-b2203bf73f0e

This is largely related to exception handling and results in some issues with tracing the control-flow of the program. It may warrant some more investigation as to the root cause of this.

Reply:
Thanks for capturing our OOB conversation @esc, as noted, no change needed to `lasti`.